template.tpl

```diff
@@ -0,0 +1,459 @@
+                  {
+                    "type": 1,
+                    "string": "read"
+                  },
+                  {
+                    "type": 1,
+                    "string": "write"
+                  },
+                  {
+                    "type": 1,
+                    "string": "execute"
+                  }
+                ],
+                "mapValue": [
+                  {
+                    "type": 1,
+                    "string": "FingerprintjsProGTM.load"
+                  },
+                  {
+                    "type": 8,
+                    "boolean": false
+                  },
+                  {
+                    "type": 8,
+                    "boolean": false
+                  },
+                  {
+                    "type": 8,
+                    "boolean": true
+                  }
+                ]
+              },
+              {
+                "type": 3,
+                "mapKey": [
+                  {
+                    "type": 1,
+                    "string": "key"
+                  },
+                  {
+                    "type": 1,
+                    "string": "read"
+                  },
+                  {
+                    "type": 1,
+                    "string": "write"
+                  },
+                  {
+                    "type": 1,
+                    "string": "execute"
+                  }
+                ],
+                "mapValue": [
+                  {
+                    "type": 1,
+                    "string": "dataLayer"
+                  },
+                  {
+                    "type": 8,
+                    "boolean": true
+                  },
+                  {
+                    "type": 8,
+                    "boolean": true
+                  },
+                  {
+                    "type": 8,
+                    "boolean": false
+                  }
+                ]
+              }
+            ]
+          }
+        }
+      ]
+    },
+    "clientAnnotations": {
+      "isEditedByUser": true
+    },
+    "isRequired": true
+  }
+]
+
+
+___TESTS___
+
+scenarios:
+- name: check loading
+  code: |-
+    const mockData = {
+      apiKey: 'aspodkasodk',
+    };
+
+    mock('injectScript', function(url, onSuccess, onFail) {
+      onSuccess();
+    });
+
+    mock('callInWindow', function(fname, getOptions, loadOptions, callback) {
+      callback({visitorId: 'qwerty'});
+    });
+
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    assertApi('injectScript').wasCalled();
+    assertApi('createQueue').wasCalled();
+    // Verify that the tag finished successfully.
+    assertApi('gtmOnSuccess').wasCalled();
+- name: check loadOptions ang getOptions with minimum params
+  code: |-
+    const mockData = {
+      apiKey: 'aspodkasodk',
+    };
+
+    const expectedLoadOptions = {
+      apiKey: 'aspodkasodk',
+      integrationInfo: 'fingerprintjs-pro-gtm-template/0.0.1',
```

Comment:
Will this fail when we update the template version?
edit: I assume it won't since it uses mocked data later.

Reply:
It will, but I don't know better solution =(